deeprank_gnn/preprocess.py

```diff
@@ -101,12 +101,24 @@ def start(self):
             if not process.is_alive():
                 raise RuntimeError(f"worker process {process.name} did not start")
 
+    def _queue_empty(self):
+        """
+        Returns whether queue is empty.
+
+        Note: This implementation is not reliable, especially not on Mac OSX. .qsize() seems to be more reliable than
+        .empty(), however qsize relies on sem_getvalue() which is not implemented on Mac OSX.
+        """
+        try:
+            return self._input_queue.qsize() == 0
+        except NotImplementedError:
+            return self._input_queue.empty()
```

Comment:
Are you sure that the preprocessing still works on a MAC, if "empty" is used?

Reply:
Well, it works, as in: it runs. But it is not reliable, as in: we don't wait long enough for all processes to finish. (I hope that is clear from the documentation that I added, if not, let me know how I can improve it). 

But I think it would be good to **properly** fix this with #56. The question is mostly: how much do you care about Mac OSX users haha.

Reply:
I suppose we don't have other options right now. Let's just merge this as soon as the checks complete.